proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+
+#### Alternative Considered: Commit Access
+
+Alternatively, the _voting contributor_ base could be defined by commit access. This methodology has two flaws. One is that our commit access list gets stale, and is unwieldy to maintain. The other problem is that with the move to GitHub PRs, commit access is significantly less important. We may begin getting more contributions from users that never bother to request commit privileges.
+
+
+### Area Teams
+
+The second structure this proposal establishes is a collection of _area teams_. Members of _area teams_ are not required or expected to be experts in the area the team is responsible for. There are no skill or experience requirements to be on an _area team_.
+
+_Area teams_ have three main responsibilities.
+
+First, they are responsible for electing from among themselves a team secretary who will take notes of any team meetings and a team chair who represents the team on the _project council_.
+
+Second, _area teams_ are responsible for maintaining an up-to-date and comprehensive list of code owners for their area of the project. They can nominate any individual they deem appropriate as code owner of any area they are responsible for. The role of _code owner_ remains a volunteer role, and any individual can accept, decline, or resign the role for themselves as they feel appropriate.
+
+Finally, _area teams_ are responsible for facilitating decision making for their area of the project. Facilitating decision making can take any number of forms ranging from contributing to RFC discussions, helping mediate disagreements, or fulfilling roles originally delegated to Chris Lattner in the [LLVM Decision Making](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md) process.
+
+When acting to facilitate decision making the _area team _should act as a mediator between different perspectives helping find common ground and recognizing that decisions need not be binary. The _area team_ should seek to find the best solution to the framed problem, which may not be any of the proposed alternatives. If agreement cannot be reached, the _area team_ may act as the final decision maker. In that capacity decisions of an _area team_ are considered final, but can be overruled by a 2/3 majority vote of the _project council_ or the _area team_ itself revisiting the issue.
+
+_Area teams_ are not intended to be direction setters or primary maintainers of their areas, although individuals on an _area team_ may fulfill that role separately. The _area team’s_ role is as a steward and moderator ensuring the health and smooth operation of the area.
+
+Each _area team_ will have 5 members elected by the _voting contributors_. Candidates for _area teams_ must be a _voting contributor_ and self-nominated. An individual cannot serve on two _area teams_.
```

Comment:
The goal here is to have a diverse leadership body. Having one person sitting on multiple area teams reduces opportunities for other people in the community and doesn't necessarily help the health of the community.

Reply:
In my _personal_ view, the health of the community is as important as the health of the project, not _more_.

In some cases, there may not be another person that can represent a project, and we'd be forced to put some random person on the team because of this restriction.

I'd make it _undesirable_, not _forbidden_.

Reply:
> Some distinct llvm components have few contributors and many more contributors who are either one-time contributors or do not want to get involved more than submitting a bug report or a patch. Do you expect most of the project areas to be able to elect and maintain healthy area teams over time?

The goal in defining the areas is that they already have enough active contributors to do this today. This is part of why the metrics I used for defining the areas wasn't just the number of contributors, but also tried to take into account the number of "highly active" contributors. There wasn't a rigorous methodology used, so it is a bit hand-wavy.